### ISSUE TITLE
Feature request: Possible to warn on missing double quote in traps?

### ISSUE THREAD

**Issue by unhammer** [koalaman/shellcheck#2584]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this (also 0.7.0)
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related
(searched for quoting in traps, but maybe there's some more general issue that would cover it?)

I would guess this is quite difficult to do in general, but OTOH for the restricted case of `trap command EXIT` maybe it's possible adding a check?

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

tmpd=$(mktemp -dt foo.XXXXXXXXXXX)
trap 'rm -rf "${tmpd}' EXIT

```

#### Here's what shellcheck currently says:

nought

#### Here's what I wanted or expected to see:

```
I fancy it might be judicious if you were to add another double quote to your exit trap, Sir.
```

**Comment by koalaman** [comment]
ShellCheck generally tries to avoid supporting especially restricted cases because it tends to overpromise on what it can do, and it's often better to be consistent than helpful but unpredictable. This is e.g. why `eval 'x=1'` is treated as a black box like all other `eval` statements, even though it would be easy to treat it the same as `x=1`. 

However, I really like this idea, because traps are usually less dynamic, bad `trap`s might not be hit during testing (for signals), and they will not be found by `bash -n` or similar.

I implemented it and ran it on a corpus, but only found 1 case in 20,000 (most checks trigger on 1 in 1000 or so). This combined with the odd behavior that you get parsing warnings only, e.g. missing one quote but not two, and that it stops emitting warnings entirely if any part of the string is dynamic makes it more odd than useful, so I won't push it. 

Great suggestion though, thanks!

PS: A function `cleanup() { ..; }` with a `trap cleanup ..` will get you syntax highlighting and checking and everything today. Maybe ShellCheck should suggest that for overly complicated traps.
